Add prototypes for Property View and Ecore Model Editor

ndoschek opened the pull request:
- Restructure client repository

- Property View Prototype
  - Add first prototype for the diagram element property view
  - property-view dependencies (locally packaged npm packages) are temporarily stored in ext-packages
  - Add custom glsp selectionmodule (will be integrated in glsp itself later on)
  - Implement glsp selection data service: glsp selection provide semantic uris for graphical model elements
  - Provide ui schemas in dedicated directory in workspace
  - Turn off jetty debug logs for modelserver product
 
- Ecore Model Editor Prototype
  - Customize theia-tree-editor as Ecore Model Editor
  - Add gifs as tree icons (extracted from Eclipse Sirius repo)
  - Remark: Tree and form updating is not finished fully yet
    - Nested nodes not correctly updated
    - Add/remove nodes on update and via tree node buttons not fully finished
    - Form widget: theia-tree-editor does not provide asynchronous methods to be able to keep the data up-to-date

- Update diagram tool palette styling
  - Remove tweaked name editing in the SubscriptionListener
  - Remove tweaking in command executing for name editing in SubscriptionListener
  - Remark: Now the editing of names is not supported for property-view and ecore model editor anymore (name fields are disabled temporarily)

Signed-off-by: Nina Doschek <ndoschek@eclipsesource.com>

CamilleLetavernier commented on client/packages/ecore-property-view/src/browser/properties-service.ts:
```diff
@@ -0,0 +1,378 @@
+        return undefined;
+    }
+
+    protected ecoreTypeSchema = {
```
Can we move this schema into its own *.json file?

CamilleLetavernier commented on client/packages/ecore-property-view/src/browser/properties-service.ts:
```diff
@@ -0,0 +1,378 @@
+        return undefined;
+    }
+
+    protected ecoreTypeSchema = {
```
Can we move this schema into its own *.json file?

ndoschek replied:
I moved it temporarily to the new common package, such that both the property view and the tree editor can use it. It is already available to fetch via the modelserver, but I noticed severe performance issues when the complete schema is fetched. Therefore we use temporarily this subset of the schema locally.

CamilleLetavernier commented on client/packages/ecore-tree-editor/src/browser/style/ecore-tree-editor.css:
```diff
@@ -0,0 +1,85 @@
+}
+
+.eattribute {
+    background-image: url(../../../../sprotty-ecore/images/EAttribute.gif);
```
These long relative URIs feel a bit fragile. Can we improve this, e.g. by sharing resources into a different package?

ndoschek replied:
That's a good idea! I moved them to a new common package and also moved the css classes there, as they are used by both the command palette and the tree editor.

CamilleLetavernier commented on client/packages/ecore-tree-editor/src/browser/tree/tree-model-service.ts:
```diff
@@ -0,0 +1,378 @@
+        return EcoreModel.Type.name(type);
+    }
+
+    public ecoreTypeSchema = {
```
Can we move this schema to a separate file as well?

CamilleLetavernier commented on server/org.eclipse.emfcloud.ecore.glsp/src/main/java/org/eclipse/emfcloud/ecore/glsp/EcoreFacade.java:
```diff
@@ -162,6 +163,12 @@ public void initializeNotationElement(NotationElement element, EObject semanticE
 				.collect(Collectors.toList());
 	}
+	
+	public List<NotationElement> findOldShapes(EClassifier eClassifier) {
```
This method can be removed; we still need to implement a reliable way to uniquely identify model-server elements

CamilleLetavernier reviewed (commented):
I've not tested the changes yet; the comments are only based on a first high-level code review. In general, I would move all UI/Schemas definitions to their own file

CamilleLetavernier commented:
Can we change the Property Labels to match the case specified in the UI Schema, rather than always using upper case? Also, can the Checkboxes be left-aligned rather than centered?
![image](https://user-images.githubusercontent.com/16717379/106598104-96b5f780-6557-11eb-936a-ba6784043279.png)

I don't know if this is related to how EMF handles default values, but it seems that "0" is never presented in this integer-value field:

![image](https://user-images.githubusercontent.com/16717379/106598740-7470a980-6558-11eb-8583-8790d696d24e.png)

Values other than 0 work fine:

![image](https://user-images.githubusercontent.com/16717379/106598757-79355d80-6558-11eb-859c-63ec619ae387.png)

ndoschek commented:
Thank you for your review Camille!
I addressed your inline comments and adjusted the styling of the property view. In that run, I aligned the styling of the tree editor and also removed the `uppercase` styling of the labels.

Regarding the `0` value:
EMF does not save the default value in that case and therefore the modelserver delivers a literal without any value property.
But displaying zero is also an issue in the version of JsonForms that is used at the moment.
I would suggest to keep track of that issue, and as soon as we consume the property view via the actual dependency (and JSONForms is updated too) we can fix that by initializing the value property for example.

CamilleLetavernier reviewed (approved):
It looks great! Thank you

CamilleLetavernier commented:
> Regarding the `0` value:
> EMF does not save the default value in that case and therefore the modelserver delivers a literal without any value property.
> But displaying zero is also an issue in the version of JsonForms that is used at the moment.
> I would suggest to keep track of that issue, and as soon as we consume the property view via the actual dependency (and JSONForms is updated too) we can fix that by initializing the value property for example.

:+1:
